### PR TITLE
Cleanup Proposals for Repository

### PR DESCRIPTION
**Pull request by JonDouglas**
This PR cleans up the existing proposals & unifies the wiki proposals into one place.

It talks about the proposal process & paves a way for managing proposals moving forward.

This is a follow-up regarding the fostering OSS efforts to drive contribution & transparency to the community.

Please feel free to provide feedback so we can make this experience wonderful moving forward for us & the community.

See https://github.com/NuGet/Home/tree/proposals-cleanup for how this would look realistically.

### CODE REVIEW

**Review by nkolev92 — APPROVED** [review]
Looks good to me. 

We should do a follow-up to go through the archive and actually categorize it correctly. 

https://github.com/NuGet/Home/tree/proposals-cleanup/proposed/archive

Alternatively, please consider putting the archive as a top folder. It was pretty difficult for me to figure out what's what.

File: meta/README.md
Comment (nkolev92) [inline meta/README.md:9]: What about proposals for problems that have existing issues?

Comment (JonDouglas) [reply meta/README.md:9]: The template should still have the issue link in it. I think it's the same process?

**Comment by JonDouglas** [comment]
I moved all top level items to make it cleaner. Archive is not perfect as I don't know the dates nor status of many of the proposals. Will need another pass.

**Comment by zivkan** [comment]
This is much improved over what we currently have.

But I was wondering what the benefit of separate `proposed/` and `implemented/` are. It requires us to create a new PR just to move accepted proposals into implemented once they're implemented. Right now we have 13 pull requests open, the oldest is from December 2019, for a feature that is implemented and already shipped (in previews, not yet GA). If we're struggling to merge design PRs, I'm not sure we're going to have the discipline of merging them into the proposed folder first, and later move them to `withdrawn/` or `implemented/`.

Whereas, if we just put all designs under `rfc/<year>/`, and keep the `status` field at the top of the file, we can merge designs with "accepted" or "withdrawn" statuses, but not need to move or edit files after the initial merge happens (unless a spec that was merged as accepted is later withdrawn). So, a little less process.

File: meta/README.md
Comment (loic-sharma) [inline meta/README.md]:
```suggestion
The official place for discussion for a proposal is its pull request or a [GitHub Discussion](https://github.com/NuGet/Home/discussions). Anyone, both NuGet contributors and non-contributors, can participate in the discussion and ask questions and provide constructive feedback to the proposal. Please keep in mind that only NuGet contributors are able to act upon the proposal even if other users can comment.
```

File: meta/README.md
Comment (loic-sharma) [inline meta/README.md]:
```suggestion
All discussions surrounding a proposal are covered by the [.NET Foundation code of conduct](https://dotnetfoundation.org/code-of-conduct). Please keep discussions constructive, respectful, and focused on what's best for the .NET ecosystem. If discussions get heated, the NuGet team may at its own discretion moderate the discussion to enforce the code of conduct.
```

File: meta/README.md
Comment (loic-sharma) [inline meta/README.md]:
```suggestion
When a proposal is reviewed for its accuracy and relevance and no longer fits the goals of the project or is no longer a candidate for implementation, a proposal will be moved from `accepted` to `withdrawn` with a clear reason as to why it was moved. 
```

File: meta/README.md
Comment (loic-sharma) [inline meta/README.md:23]: Could we link to the corresponding folder everywhere we say `proposed`, `accepted`, `widthdrawn`, or `implemented`? This would help folks jump to where they'd like to go

**Comment by loic-sharma** [comment]
This is looks great, nice work!! What do you think of adding a short readme in the new proposal folders that quickly describes the folder and links back to the meta readme? This would have the added benefit of creating the otherwise empty `accepted` folder :)

**Review by loic-sharma — APPROVED** [review]

**Comment by JonDouglas** [comment]
@zivkan I don't have strong feelings either way. My primary concern with merging proposals into `accepted` only is that in the case the work cannot be done relatively soon or may even be rejected/withdrawn due to longer community discussions, it may make sense to have a `proposed` folder for this very purpose. 

I saw the act of a NuGet maintainer moving the proposal from `proposed` to `accepted` as a means to show we are scheduling work on it. The same being from `accepted` to `implemented` to show that it has been fully implemented awaiting release meaning it's something we can blog/release note/etc on.

I think it can be accomplished many different ways. Would love to explore other proposals if they make more sense.

Good point about the stale PRs today. I'll go and clean those soon.

**Comment by JonDouglas** [comment]
@loic-sharma Great point! I'll see what I can come up with really quick to help users understand what they are looking at.

File: meta/README.md
Comment (loic-sharma) [inline meta/README.md]: Is the "will be accepted" bit correct? What do you think of this?

```suggestion
- `accepted` - The proposal has been accepted and work will be scheduled to implement it.
```

File: meta/README.md
Comment (nkolev92) [reply meta/README.md:9]: good point. I guess I'm still not convinced we should tell people to use disc when there's so many of the topics they'd be interested in are tracked in issues.

Comment (JonDouglas) [reply meta/README.md:9]: should we say something like "create a new discussion or start the conversation in an existing issue"?

Comment (nkolev92) [reply meta/README.md:9]: Yeah, I think that's fine.

File: meta/README.md
Comment (JonDouglas) [reply meta/README.md:23]: Yeah, feel free! Otherwise we can always improve these after we get this merged.

**Comment by JonDouglas** [comment]
@loic-sharma Review again to see if this suffices!

File: accepted/README.md
Comment (loic-sharma) [inline accepted/README.md]: What do you think of linking back to the overall proposal process doc? For example:

```suggestion
When a proposal has been `accepted`, it will be scheduled for work and upon completion be moved to `implemented`. For more information, see the [NuGet proposal process](https://github.com/NuGet/Home/tree/proposals-cleanup/meta#nuget-proposal-process).
```

Comment (JonDouglas) [reply accepted/README.md]: @loic-sharma Can you do this for each README & I'll merge?

Comment (JonDouglas) [reply accepted/README.md]: Actually, nevermind. I want to merge now as promised, so I'll do it.

**Pull request merged**
